### ISSUE TITLE
Add method names as constants

### ISSUE THREAD

**Issue by ghostiam** [vektra/mockery#192]
Example:
```go
	image := new(mocks.Image)
-	image.On("Add", mock.Anything)
+	image.On(mocks.ImageMethodAdd, mock.Anything)
```

```go
type Image struct {
	mock.Mock
}

+ const (
+ 	ImageMethodAdd = "Add"
+ 	ImageMethodGet = "Get"
+ )

func (_m *Image) Add(image model.Image) (uint64, error) {}
func (_m *Image) Get(settings storage.SettingsImage) ([]model.Image, error) {}
...
```

**Comment by tutukin** [comment]
Elsewhere I saw a beautiful solution^ mock has a special field, say "EXPECT" where all method control behavior is defined:
```go
mock.EXPECT.Add(mock.Anything).Return()
```

**Comment by GhostRussia** [comment]
@tutukin In this case, the restrictions are related to the library that is used for mocks
https://github.com/stretchr/testify#mock-package

For example, in another [library](
https://github.com/petergtz/pegomock) implemented as:
```go
phoneBook := NewMockPhoneBook()

// Stubbing:
pegomock.When(phoneBook.GetPhoneNumber(AnyString())).Then(func(params []Param) ReturnValues {
	return []ReturnValue{fmt.Sprintf("1-800-CALL-%v", strings.ToUpper(params[0]))}
},
// or
pegomock.When(phoneBook.GetPhoneNumber("Tom")).ThenReturn("345-123-789")

// Prints "1-800-CALL-DAN":
fmt.Println(phoneBook.GetPhoneNumber("Dan"))
phoneBook.VerifyWasCalledOnce().GetPhoneNumber("Tom")
```

**Comment by LandonTClipp** [comment]
I like this idea! I'll tag this for consideration...

**Comment by Gevrai** [comment]
> Elsewhere I saw a beautiful solution: mock has a special field, say "EXPECT" where all method control behavior is defined:
> 
> ```go
> mock.EXPECT.Add(mock.Anything).Return()
> ```

Implemented a very similar solution to this in #396, although I had to make the EXPECT a method to get the underlying mock.

**Comment by LandonTClipp** [comment]
Closed as we now have `.EXPECT()`
